### DISCUSSION TITLE
alternative way to store report data

### DISCUSSION THREAD

**Discussion by VickyXie777** [naver/ngrinder#837] [Ideas]
@junoyoon @donggyu04 

Hi guys, in later version, will you guys provide alternative way to store report data, I mean the ".data" file? When using cluster mode, NFS such as glusterfs are required, there may take a longer time to get the files when the total amount of files and repos are increased.  For example,  we want to sync the report metrics every 5s from ngrinder to our perftest centre by making an api call, delay caused by NFS is really annoying (and calling api is not a good practice, we want the controller only focus on task management). 

What about using object storage like s3 to store report data or use kafka, Which one do you recommend?

**Comment by junoyoon** [comment]
How much perftest were performed in your installation? We have no problem with .ngrinder folder size on NFS so far even with  100,000 perftests. I think your case is special case and you have to solve this by yourself.. (for example.. scanning the file system by yourself..  or synching .ngrinder folder by rsync)

By the way.. You can run multiple ngrinders in a single machine (simple cluster mode). In this case you don't need to use NFS.

**Reply by VickyXie777** [reply]
Many thanks, we will do some modification to solve it. Based on dr deployment model, we dont want to run cluster on  single machine, but use rsync instead of nfs may be a choice, if there is no collisions on svn. 

currently we have around 10,000 perftests, and we use the glusterfs support by dba team, not dedicated for ngrinder, maybe this this why the performance is influenced, but not a big "problem", nfs can handle it.

**Reply by junoyoon** [reply]
Sorry.. rsync is not appropriate for svn serving (.ngrinder/repos folder) but is OK for the other folders.
